includes/bootstrap.php

```diff
@@ -216,6 +217,75 @@ function() {
 	1
 );
 
+
+/**
+ * Add a deactivation modal when deactivating the plugin.
+ *
+ * @since x.x.x
+ */
+function register_deactivation_modal() {
+	// Exit if deactivating plugin from sub site.
+	$screen = get_current_screen();
+	if ( ! ( ! is_multisite() || $screen->in_admin( 'network' ) ) ) {
+		return;
+	}
+
+	wp_enqueue_script( 'jquery-ui-dialog' );
+	wp_enqueue_style( 'wp-jquery-ui-dialog' );
+
+	add_action(
+		'admin_footer',
+		static function () {
+			printf(
+				'<div id="my-modal" style="display:none;"><p>%1$s</p><p>%2$s</p><p><code>%3$s</code></p><p>%4$s</p></div>',
+				esc_html__( 'Would you like to delete all Distributor data?', 'distributor' ),
+				esc_html__( 'By default, the database entries are not deleted when you deactivate Distributor. If you are deleting Distributor completely from your website and want those items removed as well, add the code below to wp-config.php:', 'distributor' ),
+				'define( \'DT_REMOVE_ALL_DATA\', true );',
+				esc_html__( 'After adding this code, the Distributor plugin data will be removed from the website database when deleting the plugin. This will not delete the posts with their metadata other than the subscription. You can review uninstall.php (in the plugin root directory) to learn more about the deleted data. After deleting the Distributor plugin, you can remove the code from the wp-config.php file. Please make sure that this action cannot be undone; take a backup before proceeding.', 'distributor' )
+			);
+		}
+	);
+
+	$modal_title                   = esc_js( esc_html__( 'Distributor Deactivation', 'distributor' ) );
+	$modal_button_title_deactivate = esc_js( esc_html__( 'Deactivate', 'distributor' ) );
+	$modal_button_title_cancel     = esc_js( esc_html__( 'Cancel', 'distributor' ) );
```

Comment:
`esc_js` is for inline attributes such as `onclick` and can cause errors for use within script tags.

```suggestion
	$modal_title                   = wp_json_encode( __( 'Distributor Deactivation', 'distributor' ) );
	$modal_button_title_deactivate = wp_json_encode( __( 'Deactivate', 'distributor' ) );
	$modal_button_title_cancel     = wp_json_encode( __( 'Cancel', 'distributor' ) );
```